src/odm_sharing/private/trees.py

```diff
@@ -485,11 +485,18 @@ def parse(rules: Union[Dict[RuleId, Rule], List[Rule]],
 
     :return: an opaque rule-tree object for query generation
     '''
+    ctx = Ctx(filename)
+
     if isinstance(rules, dict):
         rules = list(rules.values())
     org_whitelist = set(orgs)
+    orgs_in_schema = seq(rules)\
+        .filter(lambda r: r.mode == RuleMode.SHARE)\
+        .map(lambda r: r.key)\
+        .set()
 
-    ctx = Ctx(filename)
+    if not (org_whitelist <= orgs_in_schema):
+        fail(ctx, 'one or more specified orgs are not in schema')
```

Comment:
Would be nice to print the error orgs along with the list of allowed orgs.

Reply:
done